pkg/cloudprovider/cloudprovider.go

```diff
@@ -74,12 +76,20 @@ type CloudProvider struct {
 }
 
 func New(ctx awscontext.Context) *CloudProvider {
+	clusterEndpoint, err := resolveClusterEndpoint(ctx, eks.New(ctx.Session))
+	if err != nil {
+		logging.FromContext(ctx).Debugf("unable to detect the cluster endpoint, %s", err)
```

Comment:
Can this be a `fatalF`? We shouldn't silently fail here if we can't discover the endpoint